proto/discard.go

```diff
@@ -256,3 +250,96 @@ func (di *discardInfo) computeDiscardInfo() {
 
 	atomic.StoreInt32(&di.initialized, 1)
 }
+
+func discardLegacy(m Message) {
+	v := reflect.ValueOf(m)
+	if v.Kind() != reflect.Ptr || v.IsNil() || v.Elem().Kind() != reflect.Struct {
+		return
+	}
+	v = v.Elem() // Kind is Struct
+
+	for i := 0; i < v.NumField(); i++ {
+		f := v.Type().Field(i)
+		if strings.HasPrefix(f.Name, "XXX_") {
+			continue
+		}
+		vf := v.Field(i)
+		tf := f.Type
+
+		// Unwrap tf to get its most basic type.
+		var isPointer, isSlice bool
+		if tf.Kind() == reflect.Slice && tf.Elem().Kind() != reflect.Uint8 {
```

Comment:
Optional.  Capture tf.Kind() into a variable and use variable here and below.

Reply:
The value of tf is changing as the logic goes on, so this will actually end up be more confusing.